src/main/java/io/appium/java_client/AddAppiumUserAgent.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client;
+
+import io.appium.java_client.internal.Config;
+import org.openqa.selenium.BuildInfo;
+import org.openqa.selenium.Platform;
+import org.openqa.selenium.remote.http.AddSeleniumUserAgent;
+import org.openqa.selenium.remote.http.Filter;
+import org.openqa.selenium.remote.http.HttpHandler;
+
+import java.util.Locale;
+
+/**
+ * Manage Appium Client configurations.
+ */
+
+public class AddAppiumUserAgent implements Filter {
```

Comment:
it would make sense to extract this functionality into a separate PR. It has nothing to do with direct connect stuff.

Also, I would rather use nouns for class names, for example `UserAgentAdder`